monai/auto3dseg/analyzer.py

```diff
@@ -779,3 +779,152 @@ def __call__(self, data):
             d[self.stats_name] = "None"
 
         return d
+
+
+class ImageHistogram(Analyzer):
+    """
+    Analyzer to compute intensity histogram.
+
+    Args:
+        image_key: the key to find image data in the callable function input (data)
+        label_key: the key to find label data in the callable function input (data)
+
+    Examples:
+
+    .. code-block:: python
+
+        import numpy as np
+        from monai.auto3dseg.analyzer import FgImageStats
+
+        input = {}
+        input['image'] = np.random.rand(1,30,30,30)
+        input['label'] = np.ones([30,30,30])
+        analyzer = FgImageStats(image_key='image', label_key='label')
+        print(analyzer(input))
+
+    """
+
+    def __init__(self, image_key: str, stats_name: str = DataStatsKeys.IMAGE_HISTOGRAM,
+                 num_of_bins: int = 100,
+                 min_value: float = -1000,
+                 max_value: float = 1000):
+
+        self.image_key = image_key
+        self.num_of_bins = num_of_bins
+        self.min_value = min_value
+        self.max_value = max_value
+        report_format = {ImageStatsKeys.HISTOGRAM: None}
+
+        super().__init__(stats_name, report_format)
+        self.update_ops(ImageStatsKeys.HISTOGRAM, SummaryOperations())
+
+    def __call__(self, data) -> dict:
+        """
+        Callable to execute the pre-defined functions
+
+        Returns:
+            A dictionary. The dict has the key in self.report_format and value
+
+        Raises:
+            RuntimeError if the stats report generated is not consistent with the pre-
+                defined report_format.
+
+        Note:
+            The stats operation uses numpy and torch to compute max, min, and other
+            functions. If the input has nan/inf, the stats results will be nan/inf.
+        """
+
+        d = dict(data)
+
+        ndas = d[self.image_key]  # (1,H,W,D) or (C,H,W,D)
+
+        # perform calculation
+        report = deepcopy(self.get_report_format())
+
+        counts, bin_edges = np.histogram(ndas, bins=self.num_of_bins, range=(self.min_value, self.max_value))
```

Comment:
@holgerroth Thanks for the PR.
If the ndas are on "cuda" device (after the `ToDeviced` transform), `np.histogram` may cause an error:
```
TypeError: can't convert cuda:0 device type tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.
```
If that's the case, I have two suggestions that may work:
- detach the `ndas` to cpu by .cpu().numpy().
- write a `histogram` function to unify the calls from numpy and torch. Here are some [examples](https://github.com/Project-MONAI/MONAI/blob/4ad790cc5f6022a92e0585c34f2f50fb6fa7fd62/monai/transforms/utils_pytorch_numpy_unification.py#L410).

Reply:
Very good point. `ndas.cpu().numpy()` seems to work fine.

Reply:
(updated to use `monai.utils.convert_to_numpy` this is compatible with both numpy/torch input.)